cpp/src/arrow/python/type_traits.h

```diff
@@ -334,6 +377,10 @@ static inline int NumPyTypeSize(int npy_type) {
       return 4;
     case NPY_FLOAT64:
       return 8;
+    case NPY_COMPLEX64:
+      return 16;
+    case NPY_COMPLEX128:
+      return 32;
```

Comment:
This seems rather strange. Shouldn't it be 8 bytes for `NPY_COMPLEX64` and 16 bytes for `NPY_COMPLEX128`? They should be exactly twice the values for NPY_FLOAT32 and NPY_FLOAT64 respectively.

Reply:
Indeed, thanks for catching this, fixed in  https://github.com/apache/arrow/pull/10565/commits/e31d556b03f5a8eae31df9a954e9dca8a0f27f14